Lib/test/test_pathlib.py

```diff
@@ -2785,6 +2776,16 @@ def check():
             check()
 
 
+class PurePathSubclassTest(_BasePurePathTest, unittest.TestCase):
+    class cls(pathlib.PurePath):
+        pass
```

Comment:
Shouldn’t this be inside a method, to make sure success/failure/error happens properly inside a test case, not when the module is loaded?

Reply:
The `_BasePurePathTest` superclass provides the test methods that use `cls`:

```python
    def test_glob(self):
        P = self.cls
        p = P(BASE)
        given = set(p.glob("FILEa"))
        expect = set() if not os_helper.fs_is_case_insensitive(BASE) else given
        self.assertEqual(given, expect)
        self.assertEqual(set(p.glob("FILEa*")), set())
```